### PR TITLE
Add `keep_original_cols()` to all steps

### PR DESCRIPTION
**Pull request by EmilHvitfeldt**
This PR add `keep_original_cols()` to all steps that needs it.

### CODE REVIEW

**Review by simonpcouch — APPROVED** [review]
Love it. :) Thanks for the thorough + nicely patterned test cases.

I had initially been hesitant about the differing defaults but it seems this is already the case with existing steps, so all good there.

Could we run revdeps on this? I wouldn't anticipate that folks would have passed arguments this far back in the formals by position, but just to make sure. It's also otherwise hard for me to eyeball whether these do indeed preserve previous behavior. Pending those checks coming back clean, thumbs up from me!

**Comment by EmilHvitfeldt** [comment]
They are not able to pass by position past the `...`, so it wouldn't be an issue :)

File: tests/testthat/_snaps/isomap.md [hunk 1]
```diff
@@ -28,8 +28,7 @@
       Error in `step_isomap()`:
       Caused by error in `prep()`:
       ! `step_isomap` failed with error:
-      Error in do.call(.Call, args = dot_call_args) : 
-        TridiagEigen: eigen decomposition failed
+      Error : TridiagEigen: eigen decomposition failed
```
[the remark this answers is not in the document]
Comment (EmilHvitfeldt) [reply tests/testthat/_snaps/isomap.md:31]: yes, this is a breach of PR etiquette. Unreleased change

[the remark this answers is not in the document]
Comment (simonpcouch) [reply tests/testthat/_snaps/isomap.md:31]: All good! Sounds good. :)

**Comment by EmilHvitfeldt** [comment]
Revdep looks good! I'll merge

Revdep results:

# Main

```r
── CHECK ───────────────────────────────────────────────────────────────────────────────────────────── 63 packages ──
✔ applicable 0.1.0                       ── E: 0     | W: 0     | N: 0                                               
✔ additive 0.0.5                         ── E: 0     | W: 0     | N: 0                                               
✔ actxps 1.1.0                           ── E: 0     | W: 0     | N: 0                                               
✔ autostats 0.4.0                        ── E: 0     | W: 0     | N: 0                                               
✔ agua 0.1.3                             ── E: 0     | W: 0     | N: 0                                               
✔ baguette 1.0.1                         ── E: 0     | W: 0     | N: 0                                               
✔ brulee 0.2.0                           ── E: 0     | W: 0     | N: 0                                               
✔ bestNormalize 1.9.0                    ── E: 0     | W: 0     | N: 0                                               
✔ bayesian 0.0.9                         ── E: 0     | W: 0     | N: 0                                               
✔ card 0.1.0                             ── E: 0     | W: 0     | N: 1                                               
✔ bundle 0.1.0                           ── E: 1     | W: 0     | N: 0                                               
✔ correlationfunnel 0.2.0                ── E: 0     | W: 0     | N: 1                                               
✔ butcher 0.3.2                          ── E: 0     | W: 0     | N: 0                                               
✔ CSCNet 0.1.2                           ── E: 0     | W: 0     | N: 0                                               
I D2MCS 1.0.1                            ── E: 1     | W: 0     | N: 0                                               
✔ caret 6.0.94                           ── E: 0     | W: 0     | N: 0                                               
✔ cvms 1.6.0                             ── E: 0     | W: 0     | N: 0                                               
✔ easyalluvial 0.3.1                     ── E: 0     | W: 0     | N: 0                                               
✔ finetune 1.1.0                         ── E: 0     | W: 0     | N: 0                                               
✔ DALEXtra 2.3.0                         ── E: 0     | W: 0     | N: 0                                               
✔ hardhat 1.3.0                          ── E: 0     | W: 0     | N: 0                                               
✔ finnts 0.2.4                           ── E: 0     | W: 0     | N: 0                                               
✔ embed 1.1.1                            ── E: 0     | W: 0     | N: 1                                               
✔ healthcareai 2.5.1                     ── E: 2     | W: 1     | N: 1                                               
I hydrorecipes 0.0.3                     ── E: 1     | W: 0     | N: 0                                               
✔ healthyR.ai 0.0.13                     ── E: 0     | W: 0     | N: 0                                               
✔ HTRX 1.2.2                             ── E: 0     | W: 0     | N: 1                                               
✔ healthyR.ts 0.2.9                      ── E: 0     | W: 0     | N: 1                                               
✔ MLDataR 1.0.1                          ── E: 0     | W: 0     | N: 1                                               
✔ modelgrid 1.1.1.0                      ── E: 0     | W: 1     | N: 2                                               
✔ MachineShop 3.6.2                      ── E: 0     | W: 0     | N: 0                                               
✔ modeltime.ensemble 1.0.3               ── E: 0     | W: 0     | N: 1                                               
✖ modeltime 1.2.7                        ── E: 0  +1 | W: 0     | N: 1                                               
✔ palmerpenguins 0.1.1                   ── E: 0     | W: 0     | N: 0                                               
✔ modeltime.resample 0.2.3               ── E: 0     | W: 0     | N: 1                                               
✔ nestedmodels 1.0.4                     ── E: 0     | W: 0     | N: 0                                               
✔ rsample 1.1.1                          ── E: 0     | W: 0     | N: 0                                               
✔ rules 1.0.2                            ── E: 0     | W: 0     | N: 0                                               
✔ probably 1.0.2                         ── E: 0     | W: 0     | N: 0                                               
✔ shinyrecipes 0.1.0                     ── E: 0     | W: 0     | N: 1                                               
✔ sknifedatar 0.1.2                      ── E: 1     | W: 0     | N: 0                                               
✔ SomaDataIO 6.0.0                       ── E: 0     | W: 0     | N: 1                                               
✖ sparseR 0.2.2                          ── E: 0  +1 | W: 0     | N: 0                                               
✔ stabiliser 1.0.6                       ── E: 0     | W: 0     | N: 1                                               
✔ stacks 1.0.2                           ── E: 0     | W: 0     | N: 0                                               
✔ swag 0.1.0                             ── E: 0     | W: 0     | N: 0                                               
✔ tabnet 0.4.0                           ── E: 0     | W: 0     | N: 0                                               
✔ text 0.9.99.2                          ── E: 0     | W: 0     | N: 1                                               
✔ textrecipes 1.0.3                      ── E: 0     | W: 0     | N: 1                                               
✔ tfhub 0.8.1                            ── E: 1     | W: 0     | N: 0                                               
✔ themis 1.0.1                           ── E: 0     | W: 0     | N: 0                                               
✔ tidybins 0.1.0                         ── E: 0     | W: 0     | N: 1                                               
✔ tidyAML 0.0.2                          ── E: 0     | W: 0     | N: 0                                               
✔ tidymodels 1.1.0                       ── E: 0     | W: 0     | N: 0                                               
✔ tidyclust 0.1.2                        ── E: 0     | W: 0     | N: 0                                               
✔ tune 1.1.1                             ── E: 0     | W: 0     | N: 0                                               
✔ usemodels 0.2.0                        ── E: 0     | W: 0     | N: 0                                               
✔ timetk 2.8.3                           ── E: 0     | W: 0     | N: 1                                               
✔ vetiver 0.2.2                          ── E: 0     | W: 0     | N: 0                                               
✔ workboots 0.2.0                        ── E: 0     | W: 0     | N: 0                                               
✔ workflows 1.1.3                        ── E: 0     | W: 0     | N: 0                                               
✔ workflowsets 1.0.1                     ── E: 0     | W: 0     | N: 0                                               
✔ waywiser 0.4.1                         ── E: 0     | W: 0     | N: 1                                               
OK: 61                                                                                                             
BROKEN: 2
Total time: 20 min
```

# This branch

```r
── CHECK ───────────────────────────────────────────────────────────────────────────────────────────── 63 packages ──
✔ additive 0.0.5                         ── E: 0     | W: 0     | N: 0                                               
✔ agua 0.1.3                             ── E: 0     | W: 0     | N: 0                                               
✔ actxps 1.1.0                           ── E: 0     | W: 0     | N: 0                                               
✔ applicable 0.1.0                       ── E: 0     | W: 0     | N: 0                                               
✔ autostats 0.4.0                        ── E: 0     | W: 0     | N: 0                                               
✔ baguette 1.0.1                         ── E: 0     | W: 0     | N: 0                                               
✔ brulee 0.2.0                           ── E: 0     | W: 0     | N: 0                                               
✔ bestNormalize 1.9.0                    ── E: 0     | W: 0     | N: 0                                               
✔ bayesian 0.0.9                         ── E: 0     | W: 0     | N: 0                                               
✔ card 0.1.0                             ── E: 0     | W: 0     | N: 1                                               
✔ bundle 0.1.0                           ── E: 1     | W: 0     | N: 0                                               
✔ butcher 0.3.2                          ── E: 0     | W: 0     | N: 0                                               
✔ caret 6.0.94                           ── E: 0     | W: 0     | N: 0                                               
✔ correlationfunnel 0.2.0                ── E: 0     | W: 0     | N: 1                                               
✔ CSCNet 0.1.2                           ── E: 0     | W: 0     | N: 0                                               
I D2MCS 1.0.1                            ── E: 1     | W: 0     | N: 0                                               
✔ cvms 1.6.0                             ── E: 0     | W: 0     | N: 0                                               
✔ easyalluvial 0.3.1                     ── E: 0     | W: 0     | N: 0                                               
✔ DALEXtra 2.3.0                         ── E: 0     | W: 0     | N: 0                                               
✔ embed 1.1.1                            ── E: 0     | W: 0     | N: 1                                               
✔ finetune 1.1.0                         ── E: 0     | W: 0     | N: 0                                               
✔ hardhat 1.3.0                          ── E: 0     | W: 0     | N: 0                                               
✔ healthcareai 2.5.1                     ── E: 2     | W: 1     | N: 1                                               
✔ finnts 0.2.4                           ── E: 0     | W: 0     | N: 0                                               
✔ healthyR.ts 0.2.9                      ── E: 0     | W: 0     | N: 1                                               
✔ HTRX 1.2.2                             ── E: 0     | W: 0     | N: 1                                               
I hydrorecipes 0.0.3                     ── E: 1     | W: 0     | N: 0                                               
✔ healthyR.ai 0.0.13                     ── E: 0     | W: 0     | N: 0                                               
✔ MLDataR 1.0.1                          ── E: 0     | W: 0     | N: 1                                               
✔ modelgrid 1.1.1.0                      ── E: 0     | W: 1     | N: 2                                               
✔ MachineShop 3.6.2                      ── E: 0     | W: 0     | N: 0                                               
✔ modeltime 1.2.7                        ── E: 0     | W: 0     | N: 1                                               
✔ modeltime.ensemble 1.0.3               ── E: 0     | W: 0     | N: 1                                               
✔ palmerpenguins 0.1.1                   ── E: 0     | W: 0     | N: 0                                               
✔ modeltime.resample 0.2.3               ── E: 0     | W: 0     | N: 1                                               
✔ nestedmodels 1.0.4                     ── E: 0     | W: 0     | N: 0                                               
✔ rules 1.0.2                            ── E: 0     | W: 0     | N: 0                                               
✔ rsample 1.1.1                          ── E: 0     | W: 0     | N: 0                                               
✔ shinyrecipes 0.1.0                     ── E: 0     | W: 0     | N: 1                                               
✔ SomaDataIO 6.0.0                       ── E: 0     | W: 0     | N: 1                                               
✔ probably 1.0.2                         ── E: 0     | W: 0     | N: 0                                               
✔ sknifedatar 0.1.2                      ── E: 1     | W: 0     | N: 0                                               
✖ sparseR 0.2.2                          ── E: 0  +1 | W: 0     | N: 0                                               
✔ swag 0.1.0                             ── E: 0     | W: 0     | N: 0                                               
✔ stacks 1.0.2                           ── E: 0     | W: 0     | N: 0                                               
✔ tabnet 0.4.0                           ── E: 0     | W: 0     | N: 0                                               
✔ stabiliser 1.0.6                       ── E: 0     | W: 0     | N: 1                                               
✔ text 0.9.99.2                          ── E: 0     | W: 0     | N: 1                                               
✔ textrecipes 1.0.3                      ── E: 0     | W: 0     | N: 1                                               
✔ tfhub 0.8.1                            ── E: 1     | W: 0     | N: 0                                               
✔ tidyAML 0.0.2                          ── E: 0     | W: 0     | N: 0                                               
✔ tidybins 0.1.0                         ── E: 0     | W: 0     | N: 1                                               
✔ themis 1.0.1                           ── E: 0     | W: 0     | N: 0                                               
✔ tidyclust 0.1.2                        ── E: 0     | W: 0     | N: 0                                               
✔ tidymodels 1.1.0                       ── E: 0     | W: 0     | N: 0                                               
✔ tune 1.1.1                             ── E: 0     | W: 0     | N: 0                                               
✔ usemodels 0.2.0                        ── E: 0     | W: 0     | N: 0                                               
✔ vetiver 0.2.2                          ── E: 0     | W: 0     | N: 0                                               
✔ timetk 2.8.3                           ── E: 0     | W: 0     | N: 1                                               
✔ workboots 0.2.0                        ── E: 0     | W: 0     | N: 0                                               
✔ waywiser 0.4.1                         ── E: 0     | W: 0     | N: 1                                               
✔ workflows 1.1.3                        ── E: 0     | W: 0     | N: 0                                               
✔ workflowsets 1.0.1                     ── E: 0     | W: 0     | N: 0                                               
OK: 62                                                                                                             
BROKEN: 1
Total time: 20 min
```

**Pull request merged**